File: Python/programmers/etc/level2/택배상자.py
def solution(order):
    assist = []
    cnt = 0
    for i in range(1, len(order)+1):
        assist.append(i)
        while assist and assist[-1] == order[cnt]:
            cnt += 1
            assist.pop()
    return cnt


order = [4, 3, 1, 2, 5]
print(solution(order)) # 2

order = [5, 4, 3, 2, 1]
print(solution(order)) # 5

File: Java/programmers/etc/level2/택배상자.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 택배상자 {
    public int solution(int[] order) {
        int cnt = 0;
        Stack<Integer> stk = new Stack<>();
        
        for (int i=1; i<order.length+1; i++) {
            stk.push(i);
            while (!stk.isEmpty() && stk.peek() == order[cnt]) {
                cnt++;
                stk.pop();
            }
        }
        
        return cnt;
    }
}
